src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -162,43 +118,30 @@ export class VisualizeLandingPageTable extends React.Component {
     return this.getColumnSortOrder(TYPE_COLUMN_ID);
   }
 
-  getAreAllItemsSelected() {
-    const { selectedIds, items, currentPage } = this.state;
-    const pageOfItems = this.pager.getItemsOnPage(items, currentPage);
-    const pageOfItemIds = pageOfItems.map(item => item.id);
-    return areAllItemsSelected(selectedIds, pageOfItemIds);
+  areAllItemsSelected() {
+    return areAllItemsSelected(this.state, this.pager);
   }
 
-  getVisualizeColumns() {
+  renderVisualizeColumns() {
     const { selectedIds } = this.state;
     return [
-      getCheckBoxColumn(this.getAreAllItemsSelected(), selectedIds, this.onToggleItem, this.onToggleAll),
-      getTitleColumn(this.getEditUrlForItem, this.getTitleSortOrder(), this.onSortByTitle),
-      getTypeColumn(this.onSortByType, this.getTypeSortOrder()),
+      getCheckBoxColumn(this.areAllItemsSelected(), selectedIds, this.onToggleItem, this.onToggleAll),
+      getTitleColumn(this.getEditUrlForItem, this.getTitleSortOrder(), this.onSort.bind(null, TITLE_COLUMN_ID)),
+      getTypeColumn(this.onSort.bind(null, TYPE_COLUMN_ID), this.getTypeSortOrder()),
     ];
   }
 
-  onDelete = () => {
-    const { deleteItems } = this.props;
-    const { selectedIds, filter } = this.state;
-    deleteItems(selectedIds).then((didDelete) => {
-      if (didDelete) {
-        this.onFilter(filter);
-      }
-    });
-  };
-
-  getActionButtons() {
+  renderActionButtons() {
     return this.state.selectedIds.length > 0
       ? <DeleteButton onClick={this.onDelete} />
       : <CreateButtonLink href={'#' + VisualizeConstants.WIZARD_STEP_1_PAGE_PATH} />;
   }
 
-  getTableContents() {
+  renderTableContents() {
```

Comment:
I think it would be ideal to separate the "column order" concern from the "table configuration" concern. Ideally the second exposes an interface which can drive by the first. Here's an example of what I mean:

```javascript
// This gets updated when the user drags and drops a header cell.
const cellOrder = [
  "id",
  "name",
  "date",
];

const cells = {
  "id": (
    <ItemTableStringCell
      key="id"
    />
  ),
  "date": (
    <ItemTableStringCell
      key="date"
    />
  ),
  "name": (
    <ItemTableStringCell
      key="name"
    />
  ),
}

const bodyRow = (
  <ItemTableBodyRow>
    {cellOrder.map(cell => cells[cell])}
  </ItemTableBodyRow>
);
```